tests/test_ical.py

```diff
@@ -0,0 +1,326 @@
+                "Only two-letter ISO 639-1 codes are supported by iCal. "
+                "Refer to the 'ISO 639-1 Code' column at "
+                "https://www.loc.gov/standards/iso639-2/php/code_list.php for valid codes."
+            ),
+        )
+
+    def test_single_holiday_event(self):
+        output = self.us_exporter.generate()
+
+        self.assertIn("BEGIN:VEVENT", output)
+        self.assertIn("DTSTAMP:", output)
+        self.assertIn("UID:", output)
+        self.assertIn("SUMMARY:New Year's Day", output)
+        self.assertIn("DTSTART;VALUE=DATE:20240101", output)
+        self.assertIn("DURATION:P1D", output)
+        self.assertIn("END:VEVENT", output)
+
+    def test_multiple_holidays_same_date(self):
+        # "วันพ่อแห่งชาติ" and "วันชาติ" are both on DEC 5th.
+        output = self.th_exporter.generate()
+
+        self.assertIn("SUMMARY:วันพ่อแห่งชาติ", output)
+        self.assertIn("SUMMARY:วันชาติ", output)
+
+    def test_single_holiday_multiple_date_continuous(self):
+        # 3x "วันสงกรานต์", 1x "ชดเชยวันสงกรานต์".
+        output = self.th_exporter.generate()
+
+        songkran_count = output.count("SUMMARY:วันสงกรานต์\r\n")
+        self.assertEqual(songkran_count, 1)
+        self.assertIn("DURATION:P3D\r\n", output)
+
+    def test_single_holiday_multiple_date_noncontinuous(self):
+        # 2x "Eid al-Fitr", 2x "Eid al-Fitr Second Day".
+        output = self.id_exporter.generate()
+
+        eid_al_fitr_count = output.count("SUMMARY:Eid al-Fitr\r\n")
+        self.assertEqual(eid_al_fitr_count, 2)
+
+    def test_combined_holidays_list(self):
+        # All 3 "New Year's Day" should be merged into 1 single instance.
+        cn_holidays = country_holidays("CN", years=2024, language="en_US")
+        jp_holidays = country_holidays("JP", years=2024, language="en_US")
+        kr_holidays = country_holidays("KR", years=2024, language="en_US")
+        east_asia_holidays = cn_holidays + jp_holidays + kr_holidays
+        output = ICalExporter(east_asia_holidays).generate()
+
+        new_years_day_count = output.count("SUMMARY:New Year's Day\r\n")
+        self.assertEqual(new_years_day_count, 1)
+
+    def test_escape_character_holiday_names(self):
+        # BACKSLASH should be include in front of COMMA.
+        output = ICalExporter(country_holidays("PT", years=2024)).generate()
+
+        self.assertIn(
+            "SUMMARY:Dia de Portugal\\, de Camões e das Comunidades Portuguesas\r\n",
+            output,
+        )
+
+    def test_localized_holiday_names(self):
+        output = self.jp_exporter.generate()
+
+        self.assertIn("SUMMARY:元日\r\n", output)
+
+    def test_empty_holidays(self):
+        exporter = ICalExporter(country_holidays("TH", years=1800))
+        output = exporter.generate()
+
+        self.assertIn("BEGIN:VCALENDAR\r\n", output)
+        self.assertIn("PRODID:", output)
+        self.assertIn("VERSION:2.0\r\n", output)
+        self.assertIn("END:VCALENDAR\r\n", output)
```

Comment:
It's useful to check here for the absence of BEGIN:VEVENT, DTSTART, SUMMARY, etc.